File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l=0
        set_1=set()
        length=0
        for i in range(len(s)):
            while s[i] in set_1:
                set_1.remove(s[l])
                l+=1
            set_1.add(s[i])
            length=max(length,i-l+1)
        return length
        

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int maxi=0;
        Set<Character> set_1=new HashSet<>();
        int l=0;
        int n=s.length();
        for(int i=0;i<n;i++)
        {
            while(set_1.contains(s.charAt(i)))
            {
                set_1.remove(s.charAt(l));
                l++;
            }
            set_1.add(s.charAt(i));
            maxi=Math.max(maxi,i-l+1);
        }
        return maxi;
    }
}